### ISSUE TITLE
[$20][MSFT-131] [Challenge Page] Prizes section is not readout by the screen reader

### ISSUE THREAD

**Issue by veshu** [topcoder-platform/community-app#2828]
**Steps to Reproduce**
1. Go to https://www.topcoder.com/challenges/30095032
2. Switch on the screen reader
3. Tap on the page header and press tab button until it reaches to the prizes section
2. Listen

**Expected Result**
It should be read-out by the screen reader

**Actual Result**
Prizes section is not read out by the screen reader

**Device**: Desktop/Labtop
**Operating System**: Windows
**Browser**: Chrome

**WCAG Category**: 1. Perceivable
**WCAG Level**: A
**WCAG Success Criterion**: 1.3.1 Info and Relationships
**Screen Reader Used**: JAWS
**Accessibility Audit Tool used**: N/A

**Attachment**
https://topcodermsft-my.sharepoint.com/personal/pd-topcoder_topcodermsft_onmicrosoft_com/Documents/Forms/All.aspx?cid=6aa53cde%2D1eef%2D4bae%2Dbfe5%2Df5f5f79e9df4&FolderCTID=0x0120005C598A51049FC14CBA882E1AEE168F51&id=%2Fpersonal%2Fpd%2Dtopcoder%5Ftopcodermsft%5Fonmicrosoft%5Fcom%2FDocuments%2FMSFT%2DTeams%2DQA%2FJune%202019%2F29%20Jun%2FTest%20Results%2FBug%20Videos%20%26%20Log%20Files%2Fcodejam%2DAccessibility%2F41

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30095960 has been created for this ticket.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by ThomasKranitsas** [comment]
Contest https://www.topcoder.com/challenges/30095960 has been updated - the new changes has been updated for this ticket.<br/><br/>```This is an automated message for thomaskranitsas via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30095960 has been updated - it has been assigned to PkDurlabhji.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@veshu PR https://github.com/topcoder-platform/community-app/pull/2949

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[500]: Challenge for the updated issue 2828 is creating, rescheduling this event<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[400]: Failed to create challenge. Detail: Operation failed in the contest service facade.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[500]: Challenge for the updated issue 2828 is creating, rescheduling this event<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[500]: Challenge for the updated issue 2828 is creating, rescheduling this event<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[504]: Failed to create challenge.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
[504]: Failed to create challenge.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30097359 has been created for this ticket.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by cwdcwd** [comment]
Contest https://www.topcoder.com/challenges/30097359 has been updated - it has been assigned to PkDurlabhji.<br/><br/>```This is an automated message for lazybaer via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@codeMinter @veshu PR updated

**Comment by PrakashDurlabhji** [comment]
@codeMinter Pr updated

**Comment by codeMinter** [comment]
@Oanh-and-only-Oanh @veshu Should this read as prizes as `tab` or `contentinfo` ? Ex: `1st prize $1000, tab` or `1st prize $1000, contentinfo` ?

**Comment by PrakashDurlabhji** [comment]
@codeMinter it should be "1st prize $1000" and nothing else. I am searching such that link or button are not called.

**Comment by veshu** [comment]
@PrakashDurlabhji @codeMinter  link or button should be called if it is link or button. They are useful info.

https://www.youtube.com/watch?v=cOmehxAU_4s 

You can check with github itself, it has great accessibility. e.g. if you go to fork number of a  repo in header. It will not just read as XX. instead it say "link, xx users forked this repository" 

So users knows this is link which is linked to xx users forks. Link, button, contentfinfo etc are useful as they provide other option for screen reader.

So provide really meaning full sentences instead of just the text there, we can see so we make correlation with other ui elements but blind won't be able relate.

My best suggestion for this would be "There are 3 prizes for this challenge, 1st prize is $1000, 2nd $500 and 3rd $250"

@codeMinter @Oanh-and-only-Oanh  regarding tab vs contentinfo. It should be contentinfo  for more detail you can check here https://www.w3.org/WAI/PF/aria/roles#role_definitions

**Comment by PrakashDurlabhji** [comment]
@veshu possible to raise amount? due to increase in efforts?

**Comment by codeMinter** [comment]
> @PrakashDurlabhji @codeMinter link or button should be called if it is link or button. They are useful info.
> 
> https://www.youtube.com/watch?v=cOmehxAU_4s
> 
> You can check with github itself, it has great accessibility. e.g. if you go to fork number of a repo in header. It will not just read as XX. instead it say "link, xx users forked this repository"
> 
> So users knows this is link which is linked to xx users forks. Link, button, contentfinfo etc are useful as they provide other option for screen reader.
> 
> So provide really meaning full sentences instead of just the text there, we can see so we make correlation with other ui elements but blind won't be able relate.
> 
> My best suggestion for this would be "There are 3 prizes for this challenge, 1st prize is $1000, 2nd $500 and 3rd $250"
> 
> @codeMinter @Oanh-and-only-Oanh regarding tab vs contentinfo. It should be contentinfo for more detail you can check here https://www.w3.org/WAI/PF/aria/roles#role_definitions

@veshu yes, I think but we cannot keep role empty, it will by default pickup considering best option, which in this case it calls it `group`, I saw it read when no role provided as `1st prize is $1000, group`

**Comment by veshu** [comment]
@codeMinter I confirmed it should be `contentinfo`
> @codeMinter @Oanh-and-only-Oanh **regarding tab vs contentinfo. It should be contentinfo** for more detail you can check here https://www.w3.org/WAI/PF/aria/roles#role_definitions

@PrakashDurlabhji Please fix if you haven't noticed above confirmation.

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30097359 has been updated - the new changes has been updated for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by veshu** [comment]
@PrakashDurlabhji prize is bumped to double, please update the status of issue and PR if you made changes. Thanks!

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30097359 has been updated - it has been assigned to PkDurlabhji.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@veshu taking to myself again as contentinfo role was already in PR. just needed minor updates

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30097359 has been updated - it has been assigned to rohitgupta_.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by PrakashDurlabhji** [comment]
@veshu @codeMinter rohit has copied my PR and nothing else.you can check it.

these both PR code are same:
https://github.com/topcoder-platform/community-app/pull/2949/files
https://github.com/topcoder-platform/community-app/pull/3157/files

@codeMinter if his code will be considered than why not mine?

**Comment by codeMinter** [comment]
@PrakashDurlabhji I don't see his PR has been reviewed yet. Even if it is same,  it will be considered as he doesn't has failing tests,  as I see in the PR.  

Plus after repeated requests you didn't care to fix PR.  If you are not responding,  PR will be Closed and ticket opened for pickup for other participants who can make exact copy of your solution making sure PR has no issues.  

This is rule and have done with other member PRs also,  not something being done by you only.

**Comment by PrakashDurlabhji** [comment]
@codeMinter but there is no difference in PR, even I can make same commit in my PR which he did.

**Comment by r0hit-gupta** [comment]
@PrakashDurlabhji sometimes a problem can only be solved in a single way. Everyone would come up with the same solution if someone asks them to solve `2+2`. Would you then claim they copied your solution?

**Comment by lakshmiathreya** [comment]
Verified Prizes read out and focused using tab. MacOS Chrome Voice Over and Win 10 Chrome NVDA.

**Comment by lakshmiathreya** [comment]
Verified on Win10/Chrome/NVDA and MacOS/Chrome/Voice Over - able to reach Prizes section via keyboard. However, in Win10/Chrome/NVDA - the prize amount is read out twice. Pls clarify @codeMinter @PrakashDurlabhji. fyi @sushilshinde @tosha5252 @drasticdpk

**Comment by crazyk07** [comment]
@sushilshinde  I'm getting the same thing as Lakshmi.  Is there a simple fix to correct this?

**Comment by crazyk07** [comment]
https://drive.google.com/file/d/1OxC67_gSRleK3lp9QZd6FjoN3b-kxgyN/view?usp=sharing

**Comment by sushilshinde** [comment]
@lakshmiathreya is it still working on test env?

**Comment by r0hit-gupta** [comment]
@lakshmiathreya @crazyk07 made a minor change. Please review.

**Comment by tosha5252** [comment]
@sushilshinde  this should be tested in the dev env or in prod?  I see it was moved to prod but it's been tagged only 'ready for QA'

**Comment by tosha5252** [comment]
@sushilshinde this works in the test env

**Comment by tosha5252** [comment]
@sushilshinde  In Win10/Chrome/NVDA..amount still reading out twice...was there a fix for this?

**Comment by veshu** [comment]
@sushilshinde Is this on production?

**Comment by tosha5252** [comment]
In prod, prize is still being read out twice using NVDA.  

WIn10/Chrome

**Comment by sushilshinde** [comment]
@crazyk07 @tosha5252 @lakshmiathreya 

Same commit 

`https://github.com/topcoder-platform/community-app/pull/3196/files`

is present on production 

https://github.com/topcoder-platform/community-app/blob/master/src/shared/components/challenge-detail/Header/Prizes.jsx#L35

@r0hit-gupta can you also confirm

**Comment by r0hit-gupta** [comment]
@sushilshinde yes, the code is present in the production. @tosha5252 it worked fine for you in dev env, right?

**Comment by tosha5252** [comment]
@r0hit-gupta yes, it worked in dev env

**Comment by r0hit-gupta** [comment]
@tosha5252 @veshu @crazyk07 I have made some changes. Should now work with both JAWS & NVDA properly. Please review.

**Comment by lakshmiathreya** [comment]
Seems to be working fine with Win 10/Chrome/NVDA and MacOS/Chrome/Voice Over ... @tosha5252 @drasticdpk pls move to QA Pass after JAWS test is done ...

**Comment by tosha5252** [comment]
Working with Win10/Chome/JAWS

**Comment by tosha5252** [comment]
Verified working correctly in prod

Win/Chrome/JAWS

**Comment by crazyk07** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30097359<br/><br/>```This is an automated message for crazyk via Topcoder X```
